zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/AuthorizationControllerTest.java

```diff
@@ -0,0 +1,124 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+
+@WebMvcTest(AuthorizationController.class)
+public class AuthorizationControllerTest extends RestControllerTest {
+  @MockBean private AuthorizationServices<AuthorizationRecord> authorizationServices;
+
+  @BeforeEach
+  void setup() {
+    when(authorizationServices.withAuthentication(any(Authentication.class)))
+        .thenReturn(authorizationServices);
+  }
+
+  @Test
+  void createAuthorizationShouldReturnNoContent() {
+    final var request = new AuthorizationAssignRequest();
+    request.setOwnerKey("1");
+    request.setOwnerType(AuthorizationOwnerType.USER);
+    request.setResourceKey("2");
+    request.setResourceType("resourceType");
+    request.setPermissions(List.of("permission1", "permission2"));
+
+    final var authorizationRecord = new AuthorizationRecord();
+    authorizationRecord.setOwnerKey(request.getOwnerKey());
+    authorizationRecord.setOwnerType(request.getOwnerType());
+    authorizationRecord.setResourceKey(request.getResourceKey());
+    authorizationRecord.setResourceType(request.getResourceType());
+    authorizationRecord.setPermissions(request.getPermissions());
+
+    when(authorizationServices.createAuthorization(
+            request.getOwnerKey(),
+            request.getOwnerType(),
+            request.getResourceKey(),
+            request.getResourceType(),
+            request.getPermissions()))
+        .thenReturn(CompletableFuture.completedFuture(authorizationRecord));
+
+    webClient
+        .post()
+        .uri("/v2/authorizations")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    verify(authorizationServices, times(1))
+        .createAuthorization(
+            request.getOwnerKey(),
+            request.getOwnerType(),
+            request.getResourceKey(),
+            request.getResourceType(),
+            request.getPermissions());
+  }
+
+  @Test
+  void createAuthorizationThrowsExceptionWhenServiceThrowsException() {
+    final String message = "message";
+
+    final var request = new AuthorizationAssignRequest();
+    request.setOwnerKey("1");
+    request.setOwnerType(AuthorizationOwnerType.USER);
+    request.setResourceKey("2");
+    request.setResourceType("resourceType");
+    request.setPermissions(List.of("permission1", "permission2"));
+
+    when(authorizationServices.createAuthorization(
+            request.getOwnerKey(),
+            request.getOwnerType(),
+            request.getResourceKey(),
+            request.getResourceType(),
+            request.getPermissions()))
+        .thenThrow(new CamundaServiceException(RejectionType.ALREADY_EXISTS.name()));
```

Comment:
🔧 I think it's more realistic if this exception takes a `BrokerRejection` because if we detect an already exists we would do this on the broker side. E.g.:

```suggestion
        .thenThrow(
            new CamundaServiceException(
                new BrokerRejection(
                    AuthorizationIntent.CREATE,
                    1L,
                    RejectionType.ALREADY_EXISTS,
                    "Authorization already exists")));
```